web/src/actions/apd.test.js

```diff
@@ -90,13 +90,29 @@ describe('apd actions', () => {
 
   describe('create APD', () => {
     // TODO: But for real.
-    it('uh...  does not do anything on success... we should change this', () => {
-      fetchMock.onPost('/apds').reply(200);
-      const store = mockStore();
+    it('adds the new APD to the store and switches to it on success', () => {
+      const newapd = { id: 'bloop' };
+      fetchMock.onPost('/apds').reply(200, newapd);
 
-      const expectedActions = [{ type: actions.CREATE_APD_REQUEST }];
+      const push = route => ({ type: 'FAKE_PUSH', pushRoute: route });
+      const state = {
+        apd: {
+          byId: {
+            bloop: { hello: 'world' }
+          }
+        }
+      };
+      const store = mockStore(state);
+
+      const expectedActions = [
+        { type: actions.CREATE_APD_REQUEST },
+        { type: actions.CREATE_APD_SUCCESS, data: newapd },
+        { type: actions.SELECT_APD, apd: { hello: 'world' } },
+        { type: actions.UPDATE_BUDGET, state },
+        { type: 'FAKE_PUSH', pushRoute: '/apd' }
+      ];
 
-      store.dispatch(actions.createApd()).then(() => {
+      return store.dispatch(actions.createApd({ pushRoute: push })).then(() => {
```

Comment:
If you don't `return` this promise, the test finishes before the action actually happens, so your test will fail but the testrunner will say everything passed.  Don't ask me how I found out.  😳

Reply:
😬